connector/src/yang/connector/grpc/telegraf.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import socket
+import configparser
+import subprocess
+import tempfile
+
+from pyats.connections import BaseConnection
+from pyats.easypy import runtime
+from pyats.utils.secret_strings import to_plaintext
+from unicon import Connection
+
+from . import Grpc
+
+# create a logger for this module
+log = logging.getLogger(__name__)
+
+
+class Grpc(Grpc):
+    """
+    Subclass for using Telegraf to collect telemetry
+    """
+    @property
+    def connected(self):
+        """Return True if session is connected."""
+        return True
+
+    def connect(self):
+        """
+        Configures and boots a telegraf process on the machine that executes the connect method wherin
+        telegraf is opened in a Python subprocess.
+
+        The network device is then connected via Unicon CLI and the outbound telemetry process that corresponds
+        to the booted telegraf process is configured on the device, with the CLI connection remaining open
+        """
+
+        # Allocate a random available port to localhost
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as grpc_socket:
+            grpc_socket.bind(('localhost', 0))
```

Comment:
Leave a note for # TODO: IPv6
```suggestion
            grpc_socket.bind(('127.0.0.1', 0))
```

Reply:
Actually, need to use `0.0.0.0` here as we need to bind the port for all IP addresses